components/brave_wallet_ui/constants/types.ts

```diff
@@ -505,6 +527,8 @@ export interface EthTxController {
   makeERC20TransferData: (toAddress: string, amount: string) => Promise<MakeERC20TransferDataReturnInfo>
   makeERC20ApproveData: (spenderAddress: string, amount: string) => Promise<MakeERC20ApproveDataReturnInfo>
   getAllTransactionInfo: (fromAddress: string) => Promise<GetAllTransactionInfoReturnInfo>
+  getMessageToSignFromTxData: (txMetaId: string) => Promise<GetMessageToSignFromTxDataReturnInfo>
```

Comment:
It's better to rename it to approveLedgerTransaction

Reply:
I think it will be reused for Trezor as well, let's keep current name?

Reply:
then `approveHardwareTransaction`, you also need to change the callback to be `OnGetNextNonceForLedger`

Reply:
done